File: 1318_Minimum_Flips_to_Make_a_OR_b_Equal_to_c.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        res = 0
        while a or b or c:
            ba, bb, bc = a & 1, b & 1, c & 1
            if ba | bb != bc:
                res += abs(ba + bb - bc)
            a >>= 1
            b >>= 1
            c >>= 1
        return res

File: 1318_Minimum_Flips_to_Make_a_OR_b_Equal_to_c.cpp
class Solution {
public:
    int minFlips(int a, int b, int c) {
        int ba, bb, bc, res = 0;
        while (a | b | c) {
            ba = a & 1;
            bb = b & 1;
            bc = c & 1;
            res += abs(ba + bb - bc);
            if (ba & bb & bc == 1) --res;
            a >>= 1;
            b >>= 1;
            c >>= 1;
        }
        return res;
    }
};
